### PR TITLE
Fleshed out the README for the `sjb` contents

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,43 @@
...
+The generator understand a small set of `action`s, each of which is underpinned by a Python module under [`sjb/actions/`](sjb/actions). A configuration YAML file is read in by [`sjb/generate.py`](sjb/generate.py) and used to generate a set of input variables to the [Jinja job template XML](sjb/templates/test_case.xml). Jobs can depend on a parent to reuse configuration.
+
+A typical workflow for a developer making changes to the job would look like:
+
+ - make edits to a configuration file under `sjb/config/`
+ - run `sjb/generate.sh`
+ - commit changes
+ - run `sjb/push-update-automatic.sh` once changes are approved and merged into `master`
+
+In order to test a job, it is necessary to copy a configuation file under `sjb/config/` to a new YAML file with a different name, then re-generate XML and use `sjb/push-update.sh` to push only the test job up to the server. Cleanup of these jobs post-test is still manual.
```
Comment (rhcarvalho) [inline README.md]: s/configuation/configuration

**Comment by stevekuznetsov** [comment]
ping @jupierce

File: README.md [hunk 2]
```diff
@@ -1,5 +1,40 @@
...
+A custom XML generator lives under the `sjb/` directory. This generator is meant to be a tightly scoped tool that would help us bridge the gap between monolithic scripts inside of Freestyle Jenkins Jobs and segmented Jenkins Pipelines driven by source-controlled Groovy scripts and libraries.
+
+The generator understand a small set of `action`s, each of which is underpinned by a Python module under [`sjb/actions/`](sjb/actions). A configuration YAML file is read in by [`sjb/generate.py`](sjb/generate.py) and used to generate a set of input variables to the [Jinja job template XML](sjb/templates/test_case.xml). Jobs can depend on a parent to reuse configuration.
+
+A typical workflow for a developer making changes to the job would look like:
+
+ - make edits to a configuration file under `sjb/config/`
+ - run `sjb/generate.sh`
+ - commit changes
+ - run `sjb/push-update-automatic.sh` once changes are approved and merged into `master`
```
Comment (smarterclayton) [inline README.md]: Reminder to set USERNAME and PASSWORD env vars here.

Comment (smarterclayton) [inline README.md]: And where those vars come from.

File: README.md [hunk 3]
```diff
@@ -1,5 +1,40 @@
...
+An internal [Continuous Infrastructure Jenkins instance](https://atomic-e2e-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/) indexes Jenkinsfiles in the branches of this repository, which are generated from the Jenkinsfiles that live under the `jobs/` directory. Jobs under the `jobs/build/` directory are indexed at the [`aos-cd-builds`](https://atomic-e2e-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/aos-cd-builds/) grouping, while the jobs under `jobs/cluster/` are indexed at the general [`aos-cd`](https://atomic-e2e-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/aos-cd/) grouping. A quick synopsis of these indexed jobs is as follows:
+
+|          Job Name          | Description |
+| -------------------------- | ----------- |
+| `build/scheduled/ose`      |             |
+| `build/scheduled/t-th`     |             |
+| `build/ose`                |             |
+| `build/stage-to-prod`      | Promote RPMs from the staging repositories to the production repositories. |
+| `build/make-puddle`        | Create an Atomic OpenShift puddle on `rcm-guest`. |
+| `build/openshift-scripts`  | Builds RPMs and container images for the [OpenShift Online](github.com/openshift/online) team. |
```
Comment (jupierce) [inline README.md]: At least in the preview, the link is relative to the repo and not resolving.

File: README.md [hunk 4]
```diff
@@ -1,5 +1,40 @@
...
+# aos-cd-jobs
 
-Multibranch jobs:
-- Branch prefix: build/*   : https://atomic-e2e-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/aos-cd-builds/
-- Branch prefix: cluster/* : https://atomic-e2e-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/aos-cd/ 
+This repository backs Jenkins jobs on a couple of Jenkins masters.
+
+## Jenkins pipeline definitions under `jobs/`
+ 
+An internal [Continuous Infrastructure Jenkins instance](https://atomic-e2e-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/) indexes Jenkinsfiles in the branches of this repository, which are generated from the Jenkinsfiles that live under the `jobs/` directory. Jobs under the `jobs/build/` directory are indexed at the [`aos-cd-builds`](https://atomic-e2e-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/aos-cd-builds/) grouping, while the jobs under `jobs/cluster/` are indexed at the general [`aos-cd`](https://atomic-e2e-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/aos-cd/) grouping. A quick synopsis of these indexed jobs is as follows:
```
Comment (jupierce) [inline README.md]: This is a great start. We can add now or later a description of how /Jenkinsfile in master does the splitting whenever the repo changes due to this job: https://atomic-e2e-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/aos-cd-master/ . It polls for changes in the repo and does the branch splitting when they are detected.

Comment (stevekuznetsov) [inline README.md]: I'll leave that one for @bbguimaraes

**Comment by stevekuznetsov** [comment]
@smarterclayton @jupierce updated

**Comment by smarterclayton** [comment]
LGTM

**Pull request merged**